src/ConcreteOperations/intersection.jl

```diff
@@ -950,38 +859,32 @@ A polyhedron.
 We assume that `rm` is polyhedral, i.e., has a `constraints_list` method
 defined.
 """
-function intersection(P::AbstractPolyhedron{N}, rm::ResetMap{N}) where {N<:Real}
+function intersection(P::AbstractPolyhedron, rm::ResetMap)
     return intersection(P, HPolyhedron(constraints_list(rm)))
 end
 
 # symmetric method
-function intersection(rm::ResetMap{N}, P::AbstractPolyhedron{N}) where {N<:Real}
-    return intersection(P, rm)
-end
+intersection(rm::ResetMap, P::AbstractPolyhedron) = intersection(P, rm)
 
 # more efficient version for polytopic
-function intersection(P::AbstractPolyhedron{N},
-                      rm::ResetMap{N, <:AbstractPolytope}) where {N<:Real}
+function intersection(P::AbstractPolyhedron{M},
+                      rm::ResetMap{N, <:AbstractPolytope}) where {M, N}
```

Comment:
The `M` is not used (same in the symmetric method below).

Reply:
:+1:

we'll reduce a bunch of LOCs with the `@commutative` macro from https://github.com/JuliaReach/LazySets.jl/pull/2286  (it is held back by incompability with julia v1.0 )